client/blocks/reader-post-card/compact.jsx

```diff
@@ -0,0 +1,68 @@
+/**
+ * External Dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal Dependencies
+ */
+import AutoDirection from 'components/auto-direction';
+import Emojify from 'components/emojify';
+import ReaderFeaturedVideo from 'blocks/reader-featured-video';
+import ReaderFeaturedImage from 'blocks/reader-featured-image';
+import ReaderExcerpt from 'blocks/reader-excerpt';
+import ReaderPostOptionsMenu from 'blocks/reader-post-options-menu';
+
+const CompactPost = ( { post, postByline, children, isDiscover } ) => {
+	const canonicalMedia = post.canonical_media;
```

Comment:
Yep. Feels like we need a `<FeaturedAsset>` component that provides this logic for both Standard and Compact cards. I'll followup with a PR for that.